### PR TITLE
feat: bump to commons 1.7.0; use non expiring assignment cache by default (FF-1072)

### PR DESCRIPTION
**Pull request by leoromanovsky**
Use non expiring assignment logging cache.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -34,6 +37,12 @@ export interface IClientConfig {
    * Pass a logging implementation to send variation assignments to your data warehouse.
    */
   assignmentLogger: IAssignmentLogger;
+
+  /**
+   * Assignment cache
+   */
+  assignmentCacheBehavior?: 'non-expiring' | 'expiring' | 'disabled';
+  assigmentExpiringCacheSize?: number;
```
[the remark this answers is not in the document]
Comment (aarsilv) [reply src/index.tsx:45]: Agree that for react native no reason to expose this other than being consistent with our node SDK.

Also agree a more explicit name would be good. `assignmentLoggingCache`?

File: src/index.tsx [hunk 2]
```diff
@@ -65,6 +74,25 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
...
+    case 'disabled':
+      EppoReactNativeClient.instance.disableAssignmentCache();
+      break;
+    case 'expiring':
+      const cacheSize =
+        config.assigmentExpiringCacheSize ||
+        DEFAULT_EXPIRING_ASSIGNMENT_CACHE_SIZE;
+      EppoReactNativeClient.instance.useLRUAssignmentCache(cacheSize);
+      break;
+    case 'non-expiring':
```
Comment (aarsilv) [inline src/index.tsx:91]: should this also be our `default:` behavior? It's an edge case as we _shouldn't_ ever get there, but I do like defensive coding. What happens if nothing matches and we don't call any cache functions?

Comment (leoromanovsky) [reply src/index.tsx:91]: yea thanks I had the same thought - wish this compiler enforced it

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@eppo/react-native-sdk",
-  "version": "0.3.3",
+  "version": "1.0.0",
```
Comment (leoromanovsky) [inline package.json:3]: I have a pet peeve about projects below version 1, makes it seem "beta"

**Review by sameerank — APPROVED** [review]

**Pull request merged**